src/upgradeExecutorFetchPrivilegedAccounts.ts

```diff
@@ -0,0 +1,157 @@
+import { Address, PublicClient } from 'viem';
+import { AbiEvent } from 'abitype';
+
+/**
+ * This type is for the params of the upgradeExecutorFetchPrivilegedAccounts function
+ */
+export type UpgradeExecutorFetchPrivilegedAccountsParams = {
+  upgradeExecutorAddress: Address;
+  publicClient: PublicClient;
+};
+
+/**
+ * This type is for the result of the upgradeExecutorFetchPrivilegedAccounts function.
+ *
+ * It is an object containing the addresses of the privileged accounts as keys,
+ * and an array with a hash for each role they have.
+ */
+export type UpgradeExecutorPrivilegedAccounts = {
+  // Key: account
+  // Value: array of roles
+  [key: `0x${string}`]: `0x${string}`[];
+};
+
+type RoleGrantedLogArgs = {
+  role: `0x${string}`;
+  account: `0x${string}`;
+  sender: `0x${string}`;
+};
+type RoleRevokedLogArgs = RoleGrantedLogArgs;
+
+const RoleGrantedEventAbi: AbiEvent = {
+  inputs: [
+    {
+      indexed: true,
+      internalType: 'bytes32',
+      name: 'role',
+      type: 'bytes32',
+    },
+    {
+      indexed: true,
+      internalType: 'address',
+      name: 'account',
+      type: 'address',
+    },
+    {
+      indexed: true,
+      internalType: 'address',
+      name: 'sender',
+      type: 'address',
+    },
+  ],
+  name: 'RoleGranted',
+  type: 'event',
+};
```

Comment:
If you remove the type on line 31, and add `as const satisfies AbiEvent;`, then Viem is able to infer the type for the args.